test/unit/generator/content/schedule/part2/index.test.js

```diff
@@ -4,9 +4,15 @@ const getTopUpDescription = require('../../../../../../app/generator/content/sch
 jest.mock('../../../../../../app/generator/content/schedule/part2/get-reduction-description')
```

Comment:
Can we better structure this test file?

We have 4 (should only be 3 as-is) scenarios to test for:
1.  top-up
2. reduction
3. zero value reduction
4. (shouldn't be) recovery]

For each of these we pass in their respective schedule and expect 2 things:
1. calls its expected function
2. doesn't call any of the others

Let's group this file into tht format

describe('top-up')
calls func
doesn't call funcA
doesn't call funcB
etc

describe('reduction')
calls func
doesn't call funcA
doesn't call funcB
etc

Reply:
Also, the functions in this aren't complete.

For each of ensuring we call top-up function we aren't checking:
1. That its called once
2. What its called with

Please add

Reply:
Finally, we don't have test cases when schedule being passed in is not any of these, i.e. undefined, null, {}, etc

Reply:
Same for part3

Reply:
The code block used is strict switch (switch(true)), hence any condition not listed in switch block would totally be ignored.